app/models/charge_mode.rb

```diff
@@ -0,0 +1,12 @@
+class ChargeMode
+
+  def self.for_order_detail(order_detail)
+    if order_detail.price_policy
+      order_detail.price_policy.charge_for
+    else
+      current_policy = order_detail.product.current_price_policies.first
+      current_policy.try(:charge_for)
+    end
+  end
+
+end
```

Comment:
Fair question. The initial reason was simply that `OrderDetail` is giant enough as is. Later reasons revolve around the fact that it's not really the order detail's responsibility. If there's a price policy assigned, then it could delegate easily enough, but when there's not one assigned, it almost becomes the product's responsibility. Since who was responsible based on the state, I decided to give it its own class.

If we had a `NullPricePolicy`, I could see some of the logic going there, but adding that in feels pretty dangerous at this point: there are a lot of checking on the presence of `price_policy`. If only you could make a null object falsy...